src/jit/codegencommon.cpp

```diff
@@ -7595,7 +7651,11 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper /*= CORINFO_HELP_PROF_FC
     // We use R8 here. This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov r8, helper addr; call r8"
+#if !defined(UNIX_AMD64_ABI)
     genEmitHelperCall(helper, 0, EA_UNKNOWN, REG_ARG_2);
+#else
+    genEmitHelperCall(helper, 0, EA_UNKNOWN, REG_R11);
+#endif
```

Comment:
@sivarv I guess that it would be better to use r10 & r11 for intermediate computations on Linux